### ISSUE TITLE
Old recorded master coverage 

### ISSUE THREAD

**Issue by ManInGotham** [jenkinsci/github-pr-coverage-status-plugin#35]
12:46:56 Master coverage +29%

Later when run on a pull request branch
14:53:59 [GitHub PR Status] master coverage: 0.27715
14:53:59 [GitHub PR Status] collecting coverage...
14:54:00 [GitHub PR Status] build coverage: 0.28635
14:54:00 [GitHub PR Status] Coverage 29% changed +0.92% vs master 28% 

Where does it take master 28%? It should be 29%. I think it compares to some old master records.

**Comment by ManInGotham** [comment]
@terma Do you know why Master Coverage might be stale? Is there a way to get detailed logs to see if it takes the latest data when comparing?

**Comment by terma** [comment]
@ManInGotham will check today

**Comment by terma** [comment]
@ManInGotham could you please check what do you have inside of ```com.github.terma.jenkins.githubprcoveragestatus.Configuration.xml``` some where in ```$JENHINS_HOME``` it should contains ```<coverageByRepo></coverageByRepo>``` where you can find master record.

**Comment by ManInGotham** [comment]
@terma Yep, I found why! There are multiple entries in Configuration.xml file.

  <coverageByRepo class="concurrent-hash-map">
    <entry>
      <string>https://company.my/repo/branchA.git</string>
      <float>0.27715</float>
    </entry>
    <entry>
      <string>https://company.my/repo/branchB.git</string>
      <float>0.286</float>
    </entry>
  </coverageByRepo>
...
I believe it happened because my job is configured with Multiple SCMs. So when it Records Master, it updates the coverage under branchB but compares with branchA values. 
How does the plugin choose which git repo to update master for?

**Comment by ManInGotham** [comment]
@terma Yep, I found why it's happening! 

There are multiple entries in Configuration.xml file.
https://company.my/org/repoA.git 
0.27715 
https://company.my/org/repoB.git 
0.286 
... 
I believe it happened because my job is configured with Multiple SCMs. When it Records Master, it updates the coverage under repoB but compares with repoA values. How does the plugin choose which git repo to update master for?

**Comment by terma** [comment]
@ManInGotham ```MasterCoverageAction``` looking for ```GIT_URL``` or ```CHANGE_URL``` to store coverage value. In your case stored master coverage and ```compare``` do they try to use totally different values or just specific branches?
